sites/zappa/app/controllers.py

```diff
@@ -45,63 +63,80 @@ def __init__(self):
     def get_file_from_repo(self, file_path: os.PathLike) -> requests.Response:
         """ get a file from a repo, file_path is from root of repository"""
         encoded_file_path = urllib.parse.quote(file_path, safe='')
-        gl_file_path = f'/projects/{self.gitlab_project_id}/repository/files/{encoded_file_path}/raw'
-        url = f'{self.gitlab_api_url}{gl_file_path}'
+        gl_file_path = f'/api/v4/projects/{self.gitlab_project_id}/repository/files/{encoded_file_path}/raw'
+        url = f'{self.gitlab_url}{gl_file_path}'
         parameters = {"ref": self.ci_branch}
         resp = requests.get(url=url, headers=self.access_headers, params=parameters)
         return resp
 
+    def create_branch(self,service_account_name):
+        gl_branch_path = f'/projects/{self.gitlab_project_id}/repository/branches'
+        parameters = {"branch": self._new_branch_name,
+                      "ref": self.ci_branch}
+        url = f'{self.gitlab_url}{gl_branch_path}'
+        try:
+            r = requests.post(url=url,headers=self.access_headers,params=parameters)
+            r.raise_for_status()
+        except requests.exceptions.HTTPError as err:
+            print(err)
+            exit(1)
+        return r.json()
+
+    def get_gitlab_access_token(self):
+        secret_store = os.getenv('FUS_SECRETS_STORE')
+        secret_name = os.getenv('GITLAB_ACCESS_KEY_SECRET_NAME')
+        return sm_client.get_secret_value(SecretId=f'{secret_store}/{secret_name}').get('SecretString')
+
+
     # https://docs.gitlab.com/ee/api/commits.html
     def commit_changes_(self, service_account_name, modified_file: FileController):
         gl_commit_path = f'/projects/{self.gitlab_project_id}/repository/commits'
         commit_message = f'This commit was created from DCP-Fusillade'
-        actions = [{'action': 'update',
+        actions = [{'action':'update',
                     'file_path': modified_file.file_path,
-                    'content': json.dumps(modified_file.updated_data, indent=4)}]
+                    'content': modified_file.updated_data}]
```

Comment:
It looks like this commit might have reverted some fixes to the Gitlab controller

Reply:
commit 5c6120d

Reply:
see #24